### PR TITLE
[ASan][X86] Mark asan_globals section large

### PR DESCRIPTION
**Pull request by aeubanks**
We'd like to make the asan_globals section large to make it not
contribute to relocation pressure since there are no direct PC32
references to it.

Following #74498, we can do that by marking the code model for the
global explicitly large.

Without this change, asan_globals gets placed between .data and .bss.
With this change, it gets placed after .bss.

### CODE REVIEW

**Review by rnk — COMMENTED** [review]
Do we need a cl::opt to control this, or is linker support for SHF_X86_64_LARGE sufficiently widespread that we don't need to worry? @MaskRay

**Comment by aeubanks** [comment]
I imagine linkers that don't support SHF_X86_64_LARGE would just pass it through without doing anything about it, which is fine.

**Comment by aeubanks** [comment]
using bfd still places asan_globals between .data and .bss while passing through SHF_X86_64_LARGE, so that seems right

**Review by vitalybuka — APPROVED** [review]

File: llvm/lib/Transforms/Instrumentation/AddressSanitizer.cpp [hunk 1]
```diff
@@ -2144,6 +2144,11 @@ ModuleAddressSanitizer::CreateMetadataGlobal(Module &M, Constant *Initializer,
       M, Initializer->getType(), false, Linkage, Initializer,
       Twine("__asan_global_") + GlobalValue::dropLLVMManglingEscape(OriginalName));
   Metadata->setSection(getGlobalMetadataSection());
+  // Move metadata section out of the way for x86-64 ELF binaries to mitigate
```
Comment (MaskRay) [inline llvm/lib/Transforms/Instrumentation/AddressSanitizer.cpp:2147]: Place metadata in a large section for ... to mitigate ...

**Review by MaskRay — APPROVED** [review]

**Review by rnk — APPROVED** [review]
Thanks!

**Pull request merged**